201-rbac-managedidentity-maps/azuredeploy.json

```diff
@@ -0,0 +1,79 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "subscriptionId": {
+      "type": "string",
+      "defaultValue": "2498e3dd-307d-4f0d-bc19-bdc139ce0c30",
+      "metadata": {
+        "description": "Subscription ID GUID"
+      }
+    },
+     "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "location for the the resources to deploy."
+      }
+    },
+    "userAssignedIdentityName": {
+      "type": "string",
+      "metadata": {
+        "description": "the name of the Managed Identity resource."
+      }
+    },
+    "mapsAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "the name of the Azure Map Account"
+      }
+    },
+    "builtInRoleType": {
+      "type": "string",
+      "defaultValue": "Azure Maps Data Reader",
+      "metadata": {
+        "description": "Input string for built in role type"
+      }
+    },
+    "guid": {
+      "type": "string",
+      "defaultValue": "916369fe-4b07-4b79-aa53-add064735d33",
+      "metadata": {
+        "description": "Input string for new GUID associated with assigning built in role types"
+      }
+    }
+  },
+  "variables": {
+    "Azure Maps Data Reader": "[concat('/subscriptions/', parameters('subscriptionId'), '/providers/Microsoft.Authorization/roleDefinitions/', '423170ca-a8f6-4b0f-8487-9e4eb8f49bfa')]"
```

Comment:
use:

[subscriptionResourceId('Microsoft.Authorization/roleDefinitions',  '423170ca-a8f6-4b0f-8487-9e4eb8f49bfa')]

Reply:
> Address the comments and CI should work - also check the CI results and fix those as well.

Best practices on CI thinks the Variable is unused: `##[error]    Unreferenced variable: Azure Maps Data Reader`

ServicePrincipal running CI doesn't have role assignment access permissions. See the [build failure](https://dev.azure.com/azurequickstarts/azure-quickstart-templates/_build/results?buildId=2222&view=logs&jobId=275f1d19-1bd8-5591-b06b-07d489ea915a&taskId=d5f35ad2-294a-5f89-d81e-1093a05b0ee3&lineStart=29&lineEnd=29&colStart=1&colEnd=779).

@bmoore-msft do you have any recommendation on solving these? I cannot assign User Access Administrator role to the principal.

Reply:
yep - see the latest PR comments